### DISCUSSION TITLE
Testing METv12.0.0

### DISCUSSION THREAD

**Discussion by fearon10** [dtcenter/METplus#2425] [Installation] [answered]
Hi There,

Our group has begun testing MET v12.0.0. We've built the v12 beta1 container in singularity from docker hub but it looks like there are some key items missing in the data directory. For example, the wrappers directory is empty. As a test, we tried pointing to the v11 wrappers but that was unsuccessful. In addition, write_tmp_dataplane.py is referenced from data/python/pyembed/ but there is no such directory in the download. Wondering if others have had similar issues?

Thanks,
Matt

**Comment by JohnHalleyGotway** [comment]
Hi Matt,

I see you have some questions about the `dtcenter/met:12.0.0-beta1` [Docker Image](https://hub.docker.com/layers/dtcenter/met/12.0.0-beta1/images/sha256-72fe9085e36e88828f4caa7d167a285cfbaf6aa2387a82226f64fb8f07a99414?context=repo).

Let me start by pointing out an important detail about the MET Docker images. We use them for both development/testing and to distributed official releases. The [build_met_docker.sh](https://github.com/dtcenter/MET/blob/1e52421b82bdcaf6eca5e71b3fd861e246994a3c/internal/scripts/docker/build_met_docker.sh#L50) script is called to compile the MET source code when creating a Docker image. You'll find the following at the end of that script:
```
if [[ $MET_GIT_NAME == "v"* ]]; then
    cd /met; rm -rf MET-*;
fi
```
So whenever we're creating a Docker image for a tagged release that begins with 'v', we delete the top-level MET source code directory. We do that to save space. These images are meant to be *run* rather than used for *development*. We have other images to support development.

But fear not, all those runtime files actually do still exist in the container in the installed `/usr/local/share/met` directory rather than in the source code where you're probably accustomed to finding them. I ran the following Docker commands to confirm:
```
docker pull dtcenter/met:12.0.0-beta1
docker run -it --rm dtcenter/met:12.0.0-beta1 ls /usr/local/share/met
   Rscripts  climo  colortables  config  fonts  map  poly	ps  python  table_files  tc_data  version.txt
docker run -it --rm dtcenter/met:12.0.0-beta1 ls /usr/local/share/met/python/pyembed/write_tmp_dataplane.py
   /usr/local/share/met/python/pyembed/write_tmp_dataplane.py
```
Hopefully that helps clarify.

Thanks, and I'm excited to hear how your use of Apptainer goes!

**Reply by fearon10** [reply]
Hi @JohnHalleyGotway 

Thank you for the quick response and for the great tips on how to locate the runtime files within the container. However, is there a place we can download separately (not within the container) the share files that include, for example, 
share/met/python/pyembed/write_tmp_dataplane.py
The tar file downloaded from https://github.com/dtcenter/MET/releases/tag/v12.0.0-beta2 has /data/python that only contains fcst.txt and obs.txt. 

Thanks again,
Matt

**Reply by JohnHalleyGotway** [reply, marked as answer]
Oh sure. Any of the files from the MET repository can easily be downloaded from GitHub. For example, that `write_tmp_dataplane.py` file lives here:
https://github.com/dtcenter/MET/blob/v12.0.0-beta2/scripts/python/pyembed/write_tmp_dataplane.py

Clicking on the "Raw" button in the top-right corner gives you a link that can easily be downloaded using wget or curl:
```
wget https://raw.githubusercontent.com/dtcenter/MET/v12.0.0-beta2/scripts/python/pyembed/write_tmp_dataplane.py
```

Alternatively, if there's a file in an image that you want to "copy out" to your local filesystem, here's how you can do that:
```
> mkdir window
> docker run -it --rm --volume `pwd`/window:/window dtcenter/met:12.0.0-beta1 cp /usr/local/share/met/python/pyembed/write_tmp_dataplane.py /window/.
> ls window
write_tmp_dataplane.py
```
The `--volume` option mounts a directory from your local machine into a location inside the container. And then I execute a command *inside the container* to copy a file into that mounted directory. After the command is run, that file appears on my local machine in the window directory.

If you do think that there is some file missing from the installed `/usr/local/share/met` directory that's actually needed at runtime, please let me know.
